docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc

```diff
@@ -0,0 +1,173 @@
+
+== Detector facts
+
+[options="autowidth"]
+|===
+| Implementation | `org.opennms.netmgt.provision.detector.bsf.BSFDetector`
+|===
+
+== Configuration and Use
+
+.Parameters for the BSF detector
+[options="header, autowidth"]
+
+|===
+| Parameter        | Description                                                                                            | Required | Default value
+| `fileName`       | The path to the file on the OpenNMS server which contains the BSF-compatible script to run.            | optional |
+| `langClass`      | The short name of the language the script is written in. Choices: `jython`, `beanshell`, `groovy`, etc.     | optional | 
+| `bsfEngine`     a| The BSF engine to use when running the script. Choices:
+
+* `bsh.util.BeanShellBSFEngine`
+* `org.codehaus.groovy.bsf.GroovyEngine`
+* `org.apache.bsf.engines.jython.JythonEngine`                                                                                | optional | 
+| `fileExtensions` | A comma-separated list of file extensions to consider a valid script for this BSF instance.            | optional | 
+| `runType`       a| Which way to run the BSF script. Choices:
+
+* `eval` - expects the script to print the OK/NOK status for validation
+* `exec` - expects the script to manipulate the provided results hashmap directly                                           | optional | `eval`
+|===
+
+CAUTION: Avoid non-character names for parameters to avoid problems in the script languages.
+
+== Timeout and Retry
+
+The BSFDetector does not perform any timeout or retry processing on its own.
+If retry and or timeout behavior is required, you must implement it in the script itself.
+
+== Requirements for the script (run-types)
+
+Depending on the `run-type`, the script has to provide its results in different ways.
+For minimal scripts with very simple logic, `run-type` `eval` is the simple option.
+Scripts running in `eval` mode have to return a string matching one of the `status codes`.
+
+If your script is more than one line, `run-type` `exec` is required.
+Scripts running in `exec` mode need not return anything, but they have to add a `status` entry with a `status code` to the _results_ object.
+Additionally, the _results_ object can also carry a "reason":"message" entry that is used in non `OK` states.
+
+== Commonly used language settings
+
+The BSF detector supports many languages; the following table provides the required setup for commonly used languages.
+
+.BSF language setups
+[options="header, %autowidth"]
+|===
+| Language                            | lang-class  | bsf-engine                                    | required library
+| http://www.beanshell.org[BeanShell] | beanshell | `bsh.util.BeanShellBSFEngine`                 | supported by default
+| http://groovy.codehaus.org[Groovy]  | groovy  | `org.codehaus.groovy.bsf.GroovyEngine`        | +groovy-all-[version].jar+
```

Comment:
groovy link was broken. Suggest the following. Please verify. The GroovyEngine parameter is still valid according to their docs. 
```suggestion
| https://groovy-lang.org/[Groovy]  | groovy  | `org.codehaus.groovy.bsf.GroovyEngine`        | +groovy-all-[version].jar+
```

Reply:
```suggestion
| https://groovy-lang.org/[Groovy]    | groovy      | `org.codehaus.groovy.bsf.GroovyEngine`        | `groovy-all-[version].jar`
```